From 4b06d6154ab41d46d2daf24dc9bc771e2b612e07 Mon Sep 17 00:00:00 2001
From: "Bartosik, Lukasz" <Lukasz.Bartosik@cavium.com>
Date: Tue, 18 Dec 2018 18:05:15 +0300
Subject: [PATCH 0859/1051] soc: octeontx2: don't register CPT AF interrupts

Don't register CPT AF interrupts if CPT block is not
implemented on a given platform.

Change-Id: If520a2649af5e1bc968184d24afbba0ccaa5c8f6
Signed-off-by: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/1766
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu.c b/drivers/soc/marvell/octeontx2/rvu.c
index c1f95965ad88..4b8ba37dc48a 100644
--- a/drivers/soc/marvell/octeontx2/rvu.c
+++ b/drivers/soc/marvell/octeontx2/rvu.c
@@ -2548,9 +2548,11 @@ static int rvu_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (err)
 		goto err_flr;
 
-	err = rvu_cpt_register_interrupts(rvu);
-	if (err)
-		goto err_irq;
+	if (is_block_implemented(rvu->hw, BLKADDR_CPT0)) {
+		err = rvu_cpt_register_interrupts(rvu);
+		if (err)
+			goto err_irq;
+	}
 
 	err = rvu_policy_init(rvu);
 	if (err)
-- 
2.17.1

